plugins/inputs/influxdb_listener/influxdb_listener.go

```diff
@@ -0,0 +1,395 @@
+		h.server.Serve(h.listener)
+	}()
+
+	h.log.Infof("Started HTTP listener service on %s", h.ServiceAddress)
+
+	return nil
+}
+
+// Stop cleans up all resources
+func (h *InfluxDBListener) Stop() {
+	deadline := time.Now().Add(10 * time.Second)
+	ctx, cancel := context.WithDeadline(context.Background(), deadline)
+	defer cancel() // go vet requires us to call cancel even though we're waiting for the deadline
+	h.server.Shutdown(ctx)
+	h.wg.Wait()
+
+	h.log.Infof("Stopped HTTP listener service on %s", h.ServiceAddress)
+}
+
+func (h *InfluxDBListener) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	h.requestsRecv.Incr(1)
+	h.mux.ServeHTTP(res, req)
+	h.requestsServed.Incr(1)
+}
+
+func (h *InfluxDBListener) handleQuery() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.queriesServed.Incr(1)
+		// Deliver a dummy response to the query endpoint, as some InfluxDB
+		// clients test endpoint availability with a query
+		res.Header().Set("Content-Type", "application/json")
+		res.Header().Set("X-Influxdb-Version", "1.0")
+		res.WriteHeader(http.StatusOK)
+		res.Write([]byte("{\"results\":[]}"))
+	}
+}
+
+func (h *InfluxDBListener) handlePing() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.pingsServed.Incr(1)
+		verbose := req.URL.Query().Get("verbose")
+
+		// respond to ping requests
+		if verbose != "" && verbose != "0" && verbose != "false" {
+			res.WriteHeader(http.StatusOK)
+			b, _ := json.Marshal(map[string]string{"version": "1.0"}) // based on header set above
+			res.Write(b)
+		} else {
+			res.WriteHeader(http.StatusNoContent)
+		}
+	}
+}
+
+func (h *InfluxDBListener) handleDefault() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.notFoundsServed.Incr(1)
+		http.NotFound(res, req)
+	}
+}
+
+func (h *InfluxDBListener) handleWrite() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.writesServed.Incr(1)
+		// Check that the content length is not too large for us to handle.
+		if req.ContentLength > h.MaxBodySize.Size {
+			tooLarge(res)
+			return
+		}
+
+		db := req.URL.Query().Get("db")
+
+		body := req.Body
+		body = http.MaxBytesReader(res, body, h.MaxBodySize.Size)
+		// Handle gzip request bodies
+		if req.Header.Get("Content-Encoding") == "gzip" {
+			var err error
+			body, err = gzip.NewReader(req.Body)
+			if err != nil {
+				h.log.Debugf("Error decompressing request body: %v", err.Error())
+				badRequest(res, err.Error())
+				return
+			}
+			defer body.Close()
+		}
+
+		parser := influx.NewStreamParser(body)
+		parser.SetTimeFunc(h.timeFunc)
+
+		var err error
+		precisionStr := req.URL.Query().Get("precision")
+		if precisionStr != "" {
+			var precision time.Duration
+			precision = getPrecisionMultiplier(precisionStr)
+			parser.SetTimePrecision(precision)
+		}
+
+		var m telegraf.Metric
+		var parseErrorCount int
+		var lastPos int
+		lastPos = 0
```

Comment:
```suggestion
		lastPos := 0
```
and then you can remove the var.

Reply:
I wanted it to be clear that it lastPos is of type int because that's what parser.Position() returns later on.  I could leave out the assignment altogether because it defaults to zero anyway, but I wanted it to look like zeroing it initially was intentional.

To get rid of the extra line I'll change it to

```suggestion
		var lastPos int = 0
```
which does the same thing but looks like it's on purpose.